### PR TITLE
Notification modal + functionality

### PR DESCRIPTION
**Pull request by mikaelollhage**

### CODE REVIEW

File: src/components/LeftMenu.vue [hunk 1]
```diff
@@ -2,9 +2,11 @@
   <div id="leftMenu">
     <div id="notifications">
       Notifications
-      <button v-if="getNewNotifications.length > 0" id="newNotifications">{{getNewNotifications.length}}</button>
+      <button v-if="getNewNotifications.length > 0" id="newNotifications" @click="setRenderNotificationsModalTrue">{{getNewNotifications.length}}</button>
       <button id="noNewNotifications" v-else>0</button>
     </div>
+
```
Comment (Martin-Ekman) [inline src/components/LeftMenu.vue:8]: NAJ

File: src/components/LeftMenu.vue [hunk 2]
```diff
@@ -2,9 +2,11 @@
   <div id="leftMenu">
     <div id="notifications">
       Notifications
-      <button v-if="getNewNotifications.length > 0" id="newNotifications">{{getNewNotifications.length}}</button>
+      <button v-if="getNewNotifications.length > 0" id="newNotifications" @click="setRenderNotificationsModalTrue">{{getNewNotifications.length}}</button>
```
Comment (Martin-Ekman) [inline src/components/LeftMenu.vue:5]: Formattering

File: src/components/LeftMenu.vue [hunk 3]
```diff
@@ -16,19 +18,31 @@
 </template>
 
 <script>
+
```
Comment (Martin-Ekman) [inline src/components/LeftMenu.vue:21]: NAJ

File: src/components/LeftMenu.vue [hunk 4]
```diff
@@ -16,19 +18,31 @@
...
+    NotificationsPopUp
+  },
   methods: {
     renderEmptySearchPageInCenter() {
       this.$store.commit("setComponentToRenderInHomeCenter", "search");
       this.$store.commit("setSearchResults", []);
       this.$store.commit("setSearchHasBeenPerformed", false);
     },
+    setRenderNotificationsModalTrue() {
+      this.$store.commit("setRenderNotificationsModal", true)
```
Comment (Martin-Ekman) [inline src/components/LeftMenu.vue:39]: SEMIKOLON

File: src/components/NotificationsPopUp.vue [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            id="songAndVideoButtons"
+            v-if="
+              notification.SharedContentType === 'song' ||
+                notification.SharedContentType === 'video'
+            "
+          >
+            <button @click="addTrackToQueueOrPlay(notification, false)">Play now</button>
+            <button @click="addTrackToQueueOrPlay(notification, true)">Add to queue</button>
+          </div>
+
```
Comment (Martin-Ekman) [inline src/components/NotificationsPopUp.vue:28]: NAJJJJ

File: src/components/NotificationsPopUp.vue [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  name: "NotificationsPopUp",
+  methods: {
+    setRenderNotificationsModalFalse() {
+      this.$store.commit("setRenderNotificationsModal", false);
+    },
+    addTrackToQueueOrPlay(notification, addToQueue) {
+      this.$store.dispatch("getMediaObjectAndAddToQueueOrPlay", {addToQueue: addToQueue, sharedContentType: notification.SharedContentType, sharedContentId: notification.SharedContentId })
+    },
+    loadMediaContent(notification) {
+      console.log(notification.SharedContentId)
```
Comment (Martin-Ekman) [inline src/components/NotificationsPopUp.vue:50]: NAJH

File: src/store/store.js [hunk 7]
```diff
@@ -191,10 +193,40 @@ export default new Vuex.Store({
...
-      const response = await fetch(`/api/notification/${this.state.user.userId}`);
+    async getNewNotifications({ commit }) {
+      //Todo: Change back route to /api/notification/${state.user.userId}
+      const response = await fetch(`/api/notification/5`);
       const newNotifications = await response.json();
       commit("setNewNotifications", newNotifications);
+    },
+    async getMediaObjectAndAddToQueueOrPlay({ commit, dispatch }, sharedContent) {
+      let response;
+      response = await fetch(
```
Comment (Martin-Ekman) [inline src/store/store.js:261]: const?

File: src/views/Home.vue [hunk 8]
```diff
@@ -54,7 +54,7 @@ export default {
       return this.$store.getters.getCenterComponentForHome;
     },
   },
-    beforeCreated(){
+    beforeMount(){
```
Comment (Martin-Ekman) [inline src/views/Home.vue:59]: formattering

**Review by Martin-Ekman — DISMISSED** [review]

**Review by juliaalexandra — DISMISSED** [review]

**Review by NDacevic — APPROVED** [review]

**Review by juliaalexandra — APPROVED** [review]

**Pull request merged**